### PR TITLE
Add Mongo collection datanode Schema

### PR DESCRIPTION
**Pull request by trgiangdo**
Schema for MongoCollectionDataNode in PR https://github.com/Avaiga/taipy-core/pull/316

@joaoandre-avaiga Can you help me check if this is schema is correct? 
Thank you.

### CODE REVIEW

File: src/taipy/rest/api/schemas/datanode.py [hunk 1]
```diff
@@ -64,6 +64,13 @@ class SQLDataNodeConfigSchema(DataNodeConfigSchema):
     write_table = fields.String()
 
 
+class MongoCollectionDataNodeConfigSchema(DataNodeConfigSchema):
+    db_username = fields.String()
```
Comment (toan-quach) [inline src/taipy/rest/api/schemas/datanode.py:68]: do we not expose `port`, `host`, `db_name` or `collection_name`, or we can access them in the properties? If we can access them from the `properties`, then do we need `db_username`, `db_password`, etc.?

Comment (toan-quach) [inline src/taipy/rest/api/schemas/datanode.py]: do we not expose port, host, db_name or collection_name, or we can access them in the properties? If we can access them from the properties, then do we need db_username, db_password, etc.?

**Review by toan-quach — COMMENTED** [review]
LGTM, just some remarks I'm not familiar with :D

File: src/taipy/rest/api/schemas/datanode.py (hunk 1 above)
Comment (trgiangdo) [reply src/taipy/rest/api/schemas/datanode.py]: I just copy from the SQLDataNode schema.
Probably the username and password is required for security reason, not sure though ...

@joaoandre-avaiga Can you help us out here? The SQL schema probably needs an update too.

Comment (jrobinAV) [reply src/taipy/rest/api/schemas/datanode.py]: The question is: Are the schemas containing a password used in an API response body or only in an API request body?

No password should be used in a response body. but we can use them in a request body. In other words, we should be able to receive a password, but we should never send a password out of taipy.

What do you think?

Comment (trgiangdo) [reply src/taipy/rest/api/schemas/datanode.py]: Agree. But I'm not sure how to ensure that actually.

Let me read a bit more on taipy-rest and I'll get back on this, and SQLSchema too.

Comment (jrobinAV) [reply src/taipy/rest/api/schemas/datanode.py]: Up. Let s not forget about this PR.
😄 
Do we have more information about that?

Comment (trgiangdo) [reply src/taipy/rest/api/schemas/datanode.py]: I've updated SQL, SQL Table, and Mongo collection data node schema with some write-only properties.

The `custom_document` property of a Mongo collection dn is a Python class, so I just leave it out for now.

Comment (toan-quach) [reply src/taipy/rest/api/schemas/datanode.py]: I still don't really like the idea of putting the password in the schema though. Where are we getting this password from again? The user will provide the password via the config and the entity will store it right? So the end user (the one using the API) doesn't need to provide or know of the password, isn't it? If so, we can just remove this field from the schema.

Comment (trgiangdo) [reply src/taipy/rest/api/schemas/datanode.py]: As my understanding, the schema can be used to config a new data node. So username and password is required too.

Comment (jrobinAV) [reply src/taipy/rest/api/schemas/datanode.py]: I am not sure what the schema is used for. for example the schema `MongoCollectionDataNodeConfigSchema`.
Is it used to configure a Mongo Collection Data node (create a `DataNodeConfig`) or is it used to retrieve create/update/read a `MongoCollectionDataNode`?

I believe it is the second hypothesis. Am I right? 
If so, (first we need to drop the Config and rename it into 'MongoCollectionDataNodeSchema') I believe The password of an entity can be updated over time. So we need to be able to set the password. No?

Comment (trgiangdo) [reply src/taipy/rest/api/schemas/datanode.py]: Actually, now reading the code for taipy-rest again, there is no API for the user to create a totally new config, or update one.

There are API to:
- Create a data node entity from config id. This config id must be already register to the Taipy app, I believe.
- Get all data node (get the config).
- read/write a data node.

So yes, I think Toan is right, we don't need the user name and password, and some other properties in this case.

Comment (jrobinAV) [reply src/taipy/rest/api/schemas/datanode.py]: OK so I guess we all agree. We can completely remove the fields.

Thx

**Review by jrobinAV — APPROVED** [review]

**Pull request merged**